src/display.c

```diff
@@ -780,10 +780,11 @@ get_input(int prompt_position, struct key *key)
 	while (true) {
 		int delay = -1;
 
-		if (opt_refresh_mode == REFRESH_MODE_PERIODIC) {
+		if (opt_refresh_mode != REFRESH_MODE_MANUAL) {
```

Comment:
This actually makes sense to me. Initially, the watch feature was lazy to avoid triggering a lot of unnecessary work. Clearly it was too lazy.